lib/Indexer/Model/Query/Criteria/ShortNameFuzzilyMatchesTo.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Phpactor\Indexer\Model\Query\Criteria;
+
+use Phpactor\Indexer\Model\Query\Criteria;
+use Phpactor\Indexer\Model\Record;
+use Phpactor\Indexer\Model\Record\HasShortName;
+
+class ShortNameFuzzilyMatchesTo extends Criteria
+{
+    public function __construct(private string $name)
+    {
+    }
+
+    public function isSatisfiedBy(Record $record): bool
+    {
+        if (!$record instanceof HasShortName) {
+            return false;
+        }
+
+        if (!$this->name) {
+            return false;
+        }
+
+        $regex = '#' . implode('.*', array_map(preg_quote(...), str_split($this->name))) . '#i';
```

Comment:
Couldn't we move this to the constructor? That way we don't have to build the regex every time.

Reply:
Good idea